GeneratingDataset.py

```diff
@@ -2677,6 +2681,24 @@ def _collect_data(self):
     else:
       assert self.feature_extractor, "feature extraction is enabled, but no audio files are specified"
       assert isinstance(first_entry["seq_name"], str)
+    # add index to data list
+    for entry in data:
+      entry['index'] = index
```

Comment:
This is not a good idea, for multiple reasons:
* `"index"` is too ambiguous. It could mean multiple things. I first assumed is the seq index, but it's not of course.
* `"index"` looks like a key which comes from the user, but it is an auxiliary key which you add here. There should be some distinction; e.g. you could use a prefix `_`.
* It's not a good idea to have this loop here. This makes it slow for lots of entries. Basically I would avoid doing this at all.

What you actually need is the mapping ref_seq_idx -> name/path/zip index. You can simply calculate this mapping, by saving how much entries every zip has. Then you don't need this.

Reply:
I see that the name was not chosen well, but I do not see the speed issue. I think it will be negligible in comparison to the time the `eval` command will need to parse the data. But if you want I can go for an alternative.

Reply:
Hm, you are right. I did not expect this. (And I think that there is a lot of chance for CPython to optimize this. I think the `eval` should actually be faster.)
Ok, then maybe leave the logic like it is, but change the key.